packages/peregrine/lib/talons/ProductFullDetail/useProductFullDetail.js

```diff
@@ -152,12 +154,24 @@ export const useProductFullDetail = props => {
         },
         [setQuantity]
     );
+    
+    const isConfigurable = isProductConfigurable(product);
+    let productPrice = product.price.regularPrice.amount;
+    if (isConfigurable) {
+        const item = findMatchingVariant({
```

Comment:
I'd like to see this follow the pattern we've used in this file already of using a memo to get some value:

```js
const productPrice = useMemo(
  () => getProductPrice(product, optionCodes, optionSelections)
)
```

The `getProductPrice` function could be another top-level utility, it would look similar to `getMediaGalleryEntries`:

```js
const getProductPrice = (product, optionCodes, optionSelections) => {
  let productPrice = product.price.regularPrice.amount;
  
  if (isProductConfigurable(product)) {
    const item = findMatchingVariant({
      optionCodes,
      optionSelections,
      variants
    });

    productPrice = item.product.price.regularPrice.amount;
  }
  
  return productPrice;
}
```